src/OrchardCore.Modules/OrchardCore.OpenId/AdminMenu.cs

```diff
@@ -32,31 +32,40 @@ public void BuildNavigation(string name, NavigationBuilder builder)
             {
                 category.AddClass("openid").Id("openid");
 
-                var features = _shellDescriptor.Features.ToImmutableArray();
-                if (features.Any(feature => feature.Id == OpenIdConstants.Features.Server) ||
-                    features.Any(feature => feature.Id == OpenIdConstants.Features.Validation))
+                var features = _shellDescriptor.Features.Select(feature => feature.Id).ToImmutableArray();
```

Comment:
Just for my knowledge, what's your reason for using `ToImmutableArray()` instead of `ToArray()` ?

Reply:
`ImmutableArray.Contains` is "non-allocatey", unlike the LINQ extension of the same name. That's the only reason to use it in this specific case (we don't really need the immutability guarantee here).

Note: it's interesting to note that the immutable types have their own LINQ-like extensions, so even things like `features.Any(feature => feature.Id == OpenIdConstants.Features.Server)` result in no enumerator being allocated.

Reply:
Thanks.